### PR TITLE
feat: add utilisationPeriodAddRegistrationCode

### PR DESCRIPTION
**Pull request by janrembold**
Add new SDK method for `/v1/utilisation-periods/${slug}/registration-codes` API endpoint

### CODE REVIEW

File: src/rest/methods/utilisationPeriod.ts [hunk 1]
```diff
@@ -165,3 +165,24 @@ export async function utilisationPeriodCheckOutUser(
...
+export async function utilisationPeriodAddRegistrationCode(
+  client: IAllthingsRestClient,
+  utilisationPeriodId: string,
+  code: string,
+): UtilisationPeriodResult {
+  return (
+    (await client.post(
+      `/v1/utilisation-periods/${utilisationPeriodId}/registration-codes`,
+      { code },
+    )) && client.utilisationPeriodGetById(utilisationPeriodId)
```
Comment (svlapin) [inline src/rest/methods/utilisationPeriod.ts]: I'm not sure querying UP back is a good approach.
Besides expanding the scope of what the method does, it becomes less deterministic: after having a rejected promise from `utilisationPeriodGetById` due to any reason, it is not clear which request has failed and so whether reg code was created or not.

Comment (janrembold) [reply src/rest/methods/utilisationPeriod.ts]: @svlapin  You're right. I simply followed the examples above :( 
I removed the method and adapted the resulting interface... Please have a look again.

File: src/rest/methods/utilisationPeriod.ts [hunk 2]
```diff
@@ -46,6 +46,8 @@ export type UtilisationPeriodResults = Promise<
   ReadonlyArray<IUtilisationPeriod>
 >
 
+export type UtilisationPeriodInvite = Promise<IUtilisationPeriodInvite>
```
Comment (svlapin) [inline src/rest/methods/utilisationPeriod.ts]: IMO a bit confusing naming here - I'd suggest to either make the `type` one end with `*Result` (for consistency reasons) or just drop it and just use `Promise<IUtilisationPeriodInvite>` as return value for `utilisationPeriodAddRegistrationCode`.
I'd prefer the latter.

Comment (janrembold) [reply src/rest/methods/utilisationPeriod.ts]: Fully right! Changed that to use only the interface

**Pull request merged**